npm/private/npm_translate_lock_helpers.bzl

```diff
@@ -0,0 +1,49 @@
+"""Starlark helpers for npm_translate_lock."""
+
+load("@bazel_skylib//lib:new_sets.bzl", "sets")
+
+def _workspace_path(rctx, short_path, user_workspace_label):
+    """Get the absolute path in the user repository of a short path using a reference label from the user repository."""
+    label_path = rctx.path(user_workspace_label)
+    label_short_path = user_workspace_label.package + ("/" if user_workspace_label.package else "") + user_workspace_label.name
+    workspace_path = str(label_path).removesuffix(label_short_path)
+    return rctx.path(workspace_path + "/" + short_path)
+
+def _verify_patches(rctx):
+    if rctx.attr.verify_patches and rctx.attr.patches != None:
+        patches_folder_path = _workspace_path(rctx, rctx.attr.verify_patches, rctx.attr.pnpm_lock)
+
+        files_in_patch_folder = patches_folder_path.readdir()
+        all_patch_files = sets.make()
+        for file in files_in_patch_folder:
+            if file.basename.find(".") >= 0:
+                if file.basename.rsplit(".")[-1] in rctx.attr.verify_patches_extensions:
```

Comment:
incase the extension the user gave has a dot like "foo.bar" this check is more robust as
```
for e in rctx.attr.verify_patches_extensions:
    if file.basename.endsiwth("."+e):
```